proposals/0048-syscall-for-secp256r1-sigverify.md

```diff
@@ -0,0 +1,86 @@
+---
+simd: '0048'
+title: Syscall for verifying secp256r1 signatures
+authors:
+  - SirIsaac (Bunkr)
+  - Jstnw (Bunkr)
+category: Standard
+type: Core
+status: Draft
+created: 2023-05-14
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+Adding a Syscall to support the verification of sigatures generated on the secp256r1 curve. 
+Similar to the support for secp256k1 signatures that already exists in form of the `KeccakSecp256k11111111111111111111111111111` native program an the `secp256k1_recover()` syscall.
+
+## Motivation
+
+Solana should have option to secure your funds in a self custodial manner that doesn't just airgap your private key with a hardware wallet (which even then remains as a single point of failure). Arguably, multi-signature wallets fit into this equation as they enable the dependency on multiple private keys. However in practice the UX takes too much of a hit, as having to sign a transaction a minimum of 3 seperate times and having to write down 3 seed phrases is too cumbersome. It would be ideal to have an authetication form that relies on a more familiar second factor, such as a users mobile device.
+
+Passkeys & WebAuthn are a standardized implementation of this. They enable users to save keypairs associated to different services natively on the secure element of their mobile device. To authenticate with those services, the user uses their biometrics to sign a message with the stored private key. 
+
+And although this is meant to enable password-less logins in web2, it makes for an excellent canditate as a second factor of on-chain authentication.
+
+Going past just securing funds, this would support other beneficial account abstractions that make use of the simple UX of WebAuthn and Passkeys.
+
+Note:
+
+Although WebAuthn supports the following curves:
+- P-256
+- P-384
+- P-521
+- ed25519
+
+P-256 is the only one suported by both Android & IOS (IOS being the more restrictive of the two), hence the goal being to implement secp256r1 signature verification
+
+General Documentation:
+
+[WebAuthn](https://webauthn.io/)
+
+[Passkeys](https://fidoalliance.org/passkeys/)
+
+## Alternatives Considered
+We have discussed the following alternatives:
+
+1.) Realising signature verification with a native program/instruction similar to `KeccakSecp256k11111111111111111111111111111` instead of a syscall. As the signature is known/generated by the user upfront for a tx, an implementation as a native program would also work.
```

Comment:
@lichtso - from the perspective of ABIv2, a native program/instruction would be strongly preferred over another syscall right? Or does it not really matter?

Reply:
Would there be any benefit / significant man hours saved if I went with the native program approach?

Reply:
I think we probably should hold here until @Lichtso and the runtime team decides what we're going to do in general about the "Precompile" instructions that currently exist for runtime-v2. We've had some Discord discussions about that over the last day or so.

Generally I feel like adding a syscall isn't the best approach if we decide to retain the notion of Precompiles in runtime-v2, since that limits our future ability to optimize the performance of this signature verification should it gain widespread usage. Others might argue that would be a premature optimization though..

Reply:
Gotcha, okay 👍

Reply:
Just want to ping in here again :) @Lichtso would you have anything against an implementation as a native program / do you see any up or downsides for either option?